osu.Game/Rulesets/Scoring/ScoreProcessor.cs

```diff
@@ -88,27 +90,30 @@ public class ScoreProcessor : JudgementProcessor
         private readonly double accuracyPortion;
         private readonly double comboPortion;
 
-        private int maxAchievableCombo;
+        /// <summary>
+        /// Scoring values for a perfect play.
+        /// </summary>
+        public ScoringValues MaximumScoringValues { get; private set; }
```

Comment:
As a safety, I'd hope that this would `throw` if attempting to access before `ApplyBeatmap` has been run.

```diff
diff --git a/osu.Game/Rulesets/Scoring/ScoreProcessor.cs b/osu.Game/Rulesets/Scoring/ScoreProcessor.cs
index 853f43ddb6..0f7056ec06 100644
--- a/osu.Game/Rulesets/Scoring/ScoreProcessor.cs
+++ b/osu.Game/Rulesets/Scoring/ScoreProcessor.cs
@@ -93,7 +93,18 @@ public class ScoreProcessor : JudgementProcessor
         /// <summary>
         /// Scoring values for a perfect play.
         /// </summary>
-        public ScoringValues MaximumScoringValues { get; private set; }
+        public ScoringValues MaximumScoringValues
+        {
+            get
+            {
+                if (!beatmapApplied)
+                    throw new InvalidOperationException($"Cannot access maximum scoring values without calling {nameof(ApplyBeatmap)}.");
+
+                return maximumScoringValues;
+            }
+        }
+
+        private ScoringValues maximumScoringValues;
 
         /// <summary>
         /// Scoring values for the current play assuming all perfect hits.
@@ -406,7 +417,7 @@ protected override void Reset(bool storeResults)
             lastHitObject = null;
 
             if (storeResults)
-                MaximumScoringValues = currentScoringValues;
+                maximumScoringValues = currentScoringValues;
 
             currentScoringValues = default;
             currentMaximumScoringValues = default;

```

Reply:
I dunno, it's not strictly speaking incorrect. I'll do it, but I was actually going to go in the opposite direction and remove `beatmapApplied` along with `ComputePartialScore()`.

Reply:
Maybe leave that for a follow-up PR.

Reply:
In a follow-up PR of course, as said in the OP.